File: src/app/shared/model/response/productDTO.ts
import { GenericDTO } from './GenericDTO';

export interface ProductDTO{
    id?:number;
    name?:string;
    summaryname?:string;
    isexpiratedate?:string;
    isrefrigeration?:string;
    isbatch?:string;
    isgeneric?:string;
    iskit?:string;
    ismedicine?:string;
    iscontrolled?:string;
    barcode?:string;
    category?:GenericDTO;
    mark?:GenericDTO;
    unit?:GenericDTO;
}

File: src/main/java/com/sys/botica/crce/pe/sys_botica/dto/ProductDTO.java
package com.sys.botica.crce.pe.sys_botica.dto;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ProductDTO {
	private Long id;
	private String name;
	private String summaryname;
	private String isexpiratedate;
	private String isrefrigeration;
	private String isbatch;
	private String isgeneric;
	private String iskit;
	private String ismedicine;
	private String iscontrolled;
	private String barcode;
	private GenericDTO category;
	private GenericDTO mark;
	private GenericDTO unit;
}
